languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -35450,6 +35450,11 @@ USA
             <example correction="materials|matériels">I needed more electronic <marker>materiels</marker> to build a fully-functional machine.</example>
         </rule>
         <rule id="BARCA_DIACRITIC" name="Barca or Barça">
+            <antipattern>
+                <token skip="-1" postag="SENT_START"/>
+                <token skip="-1" regexp="yes">surname|cognomen|Hannibal|Hamilcar|Hasdrubal</token>
+                <token postag="SENT_END"/>
+            </antipattern>
```

Comment:
I think this is an interesting variant on Ruud's method. I think it makes sense that some elements are forbidden in the entire sentence (`surname` and `cognomen`). For the first names and `Latin`, it should be sufficient to exclude them directly before `Barca`. The word "Latin" is especially likely to appear in a sentence where one's actually talking about Barça.